core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/prompt/PromptBuilder.kt

```diff
@@ -2,149 +2,167 @@ package org.jetbrains.research.testspark.core.generation.llm.prompt
-        if (isPromptValid(PromptKeyword.TESTING_PLATFORM, prompt)) {
-            val keyword = "\$${PromptKeyword.TESTING_PLATFORM.text}"
-            prompt = prompt.replace(keyword, testingPlatformName, ignoreCase = false)
-        } else {
-            throw IllegalStateException("The prompt must contain ${PromptKeyword.TESTING_PLATFORM.text}")
-        }
+        insert(PromptKeyword.TESTING_PLATFORM, testingPlatformName)
     }
 
     fun insertMockingFramework(mockingFrameworkName: String) = apply {
-        if (isPromptValid(PromptKeyword.MOCKING_FRAMEWORK, prompt)) {
-            val keyword = "\$${PromptKeyword.MOCKING_FRAMEWORK.text}"
-            prompt = prompt.replace(keyword, mockingFrameworkName, ignoreCase = false)
-        } else {
-            throw IllegalStateException("The prompt must contain ${PromptKeyword.MOCKING_FRAMEWORK.text}")
-        }
+        insert(PromptKeyword.MOCKING_FRAMEWORK, mockingFrameworkName)
     }
 
-    fun insertCodeUnderTest(classFullText: String, classesToTest: List<ClassRepresentation>) = apply {
-        if (isPromptValid(PromptKeyword.CODE, prompt)) {
-            val keyword = "\$${PromptKeyword.CODE.text}"
-            var fullText = "```\n${classFullText}\n```\n"
-
-            for (i in 2..classesToTest.size) {
-                val subClass = classesToTest[i - 2]
-                val superClass = classesToTest[i - 1]
-
-                fullText += "${subClass.qualifiedName} extends ${superClass.qualifiedName}. " +
-                    "The source code of ${superClass.qualifiedName} is:\n```\n${superClass.fullText}\n" +
-                    "```\n"
-            }
-            prompt = prompt.replace(keyword, fullText, ignoreCase = false)
-        } else {
-            throw IllegalStateException("The prompt must contain ${PromptKeyword.CODE.text}")
+    /**
+     * Inserts the code under test and its related superclass code into the prompt template.
+     *
+     * @param codeFullText The full text of the code under test.
+     * @param classesToTest The list of ClassRepresentation objects representing the classes involved in the code under test.
+     * @return The modified prompt builder.
+     */
+    fun insertCodeUnderTest(codeFullText: String, classesToTest: List<ClassRepresentation>) = apply {
+        var fullText = "```\n${codeFullText}\n```\n"
+
+        for (i in 2..classesToTest.size) {
+            val subClass = classesToTest[i - 2]
+            val superClass = classesToTest[i - 1]
+
+            fullText += "${subClass.qualifiedName} extends ${superClass.qualifiedName}. " +
+                "The source code of ${superClass.qualifiedName} is:\n```\n${superClass.fullText}\n" +
+                "```\n"
         }
+
+        insert(PromptKeyword.CODE, fullText)
     }
 
     fun insertMethodsSignatures(interestingClasses: List<ClassRepresentation>) = apply {
-        val keyword = "\$${PromptKeyword.METHODS.text}"
+        var fullText = ""
```

Comment:
Using StringBuilder will be more suitable

Reply:
Replaced with StringBuilder.